inc/graphql.php

```diff
@@ -0,0 +1,143 @@
+<?php
+/**
+ * This file contains functionality needed to integrate with WPGraphQL.
+ *
+ * @package Byline_Manager
+ */
+
+namespace Byline_Manager;
+
+use WPGraphQL;
+
+add_action( 'graphql_register_types', __NAMESPACE__ . '\register_byline_types' );
+
+/**
+ * Register data types and field for supporting bylines on posts.
+ */
+function register_byline_types() {
+	// Register a text profile.
+	register_graphql_object_type(
+		'TextProfile',
+		[
+			'description' => __( 'A TextProfile type', 'byline-manager' ),
+			'fields'      => [
+				'text' => [
+					'type'        => 'String',
+					'description' => __( 'Text for a text profile', 'byline-manager' ),
+				],
+			],
+		]
+	);
+
+	// Register a union for both Profile and TextProfile types.
+	register_graphql_union_type(
+		'ProfileTypes',
+		[
+			'typeNames'   => [ 'Profile', 'TextProfile' ],
+			'resolveType' => function ( $profile ) {
+				if (
+					$profile instanceof WPGraphQL\Model\Post
+					&& PROFILE_POST_TYPE === $profile->post_type
+				) {
+					return 'Profile';
+				}
+
+				return 'TextProfile';
+			},
+		]
+	);
+
+	// Register options for byline format.
+	register_graphql_enum_type(
+		'BylineTextEnum',
+		[
+			'description' => __( 'Byline format', 'byline-manager' ),
+			'values'      => [
+				'TEXT'       => [
+					'value' => 'text',
+				],
+				'LINKS'      => [
+					'value' => 'links',
+				],
+				'POST_LINKS' => [
+					'value' => 'post_links',
+				],
+			],
+		]
+	);
+
+	register_graphql_object_type(
+		'Byline',
+		[
+			'description' => __( 'Byline manager byline', 'byline-manager' ),
+			'fields'      => [
+				'bylineText' => [
+					'args'        => [
+						'format' => [
+							'type'        => 'BylineTextEnum',
+							'description' => __( 'Choose the byline format.', 'byline-manager' ),
```

Comment:
The rest of the descriptions don't have end punctuation, so for the sake of consistency, I'd suggest this shouldn't either
```suggestion
							'description' => __( 'Choose the byline format', 'byline-manager' ),
```

Reply:
Looking at the way fields are registered in the main WPGraphQL plugin, the assumption is that these should end in punctuation, so I'll update all of them.